### ISSUE TITLE
[BUG - Release] Steam Deck crashes on startup

### ISSUE THREAD

**Issue by diesel64** [UE4SS-RE/RE-UE4SS#345]
**Branch or Release**
3.0.0

**Game and Engine Version**
Persona 3 Reload
UE4 4.27

**Describe the bug**
Game crashes on startup, no errors seem to be listed in the log.

**To Reproduce**
1. Launch game

**Expected behavior**
An error popup stating "Unable to load library steamclient64.dll"

**Screenshots and UE4SS Log**
[UE4SS.log](https://github.com/UE4SS-RE/RE-UE4SS/files/14184222/UE4SS.log)

**Desktop (please complete the following information):**
 - OS: SteamOS

**Comment by Buckminsterfullerene02** [comment]
Steam deck is not officially supported by UE4SS. 

However, have you tried following this instruction?
https://www.reddit.com/r/HarryPotterGame/comments/11h4v1c/comment/jiom3hs/?utm_source=share&utm_medium=web2x&context=3

**Comment by bunkbail** [comment]
I'm getting the same exact error on Debian. Seems like UE4SS doesn't play well with P3R. I've followed the instructions on reddit (they typo'd the dll file tho) but to no avail.

**Comment by Buckminsterfullerene02** [comment]
Native linux support isn't quite here yet, so you need to do some messing about with wine.

**Comment by trumank** [comment]
> Native linux support

Is completely irrelevant as there is no Linux build of the game

**Comment by KingKrouch** [comment]
> Steam deck is not officially supported by UE4SS.
> 
> However, have you tried following this instruction? https://www.reddit.com/r/HarryPotterGame/comments/11h4v1c/comment/jiom3hs/?utm_source=share&utm_medium=web2x&context=3

I have tried this without much luck at all, I've also tried using Ultimate ASI Loader and renaming the UE4SS.dll to have an .asi extension so it would load it (in case it was an issue with how the library hooked the DLL into the process), I've tried WINEDLLOVERRIDES, and yet, nothing fixes the problem for me. Something's up with how UE4SS functions that prevents it from even running on Proton.

**Comment by UE4SS** [comment]
> > Steam deck is not officially supported by UE4SS.
> > However, have you tried following this instruction? https://www.reddit.com/r/HarryPotterGame/comments/11h4v1c/comment/jiom3hs/?utm_source=share&utm_medium=web2x&context=3
> 
> I have tried this without much luck at all, I've also tried using Ultimate ASI Loader and renaming the UE4SS.dll to have an .asi extension so it would load it (in case it was an issue with how the library hooked the DLL into the process), I've tried WINEDLLOVERRIDES, and yet, nothing fixes the problem for me. Something's up with how UE4SS functions that prevents it from even running on Proton.

While I can't comment on whether UE4SS works on the steam deck (I have no idea), I can say that it definitely works with proton, and in fact, that's I use it most of the time.

**Comment by UE4SS** [comment]
Oh and ever since 3.0, I haven't needed to use WINEDLLOVERRIDES, it just works out of the box if dwmapi.dll exists in the directory.

**Comment by KingKrouch** [comment]
Yeah, I'm using it on Proton, and even without the WINEDLLOVERRIDES, it looks like the same error regarding the steamclient64.dll happens. I've tried this with Hogwarts Legacy, Persona 3 Reload, and a few other games without any luck.

**Comment by trumank** [comment]
IIRC this had something to do with UE4SS touching the DLL search path to load cpp mods and had to comment these lines out to get a particular game to work through WINE: https://github.com/UE4SS-RE/RE-UE4SS/blob/332eea5711d036edd428b733f27214e8128e3d96/UE4SS/src/UE4SSProgram.cpp#L394-L397

**Comment by KingKrouch** [comment]
> IIRC this had something to do with UE4SS touching the DLL search path to load cpp mods and had to comment these lines out to get a particular game to work through WINE: https://github.com/UE4SS-RE/RE-UE4SS/blob/332eea5711d036edd428b733f27214e8128e3d96/UE4SS/src/UE4SSProgram.cpp#L394-L397

Do you happen to have a test build compiled so I can test this?

**Comment by UE4SS** [comment]
> > IIRC this had something to do with UE4SS touching the DLL search path to load cpp mods and had to comment these lines out to get a particular game to work through WINE: https://github.com/UE4SS-RE/RE-UE4SS/blob/332eea5711d036edd428b733f27214e8128e3d96/UE4SS/src/UE4SSProgram.cpp#L394-L397
> 
> Do you happen to have a test build compiled so I can test this?

You can try this one: https://github.com/UE4SS-RE/RE-UE4SS/releases/download/experimental/UE4SS_v3.0.1-52-g7d6f790.zip
In that one, we don't use SetDefaultDllDirectories.

**Comment by KingKrouch** [comment]
> > > IIRC this had something to do with UE4SS touching the DLL search path to load cpp mods and had to comment these lines out to get a particular game to work through WINE: https://github.com/UE4SS-RE/RE-UE4SS/blob/332eea5711d036edd428b733f27214e8128e3d96/UE4SS/src/UE4SSProgram.cpp#L394-L397
> > 
> > 
> > Do you happen to have a test build compiled so I can test this?
> 
> You can try this one: https://github.com/UE4SS-RE/RE-UE4SS/releases/download/experimental/UE4SS_v3.0.1-52-g7d6f790.zip In that one, we don't use SetDefaultDllDirectories.

Sweet, this seems to work without issues with Persona 3 Reload on Proton at least. Thanks!

**Comment by LuisMayo** [comment]
Hi

Can confirm both the error with the normal build and the experimental build fixing it in latest SteamOS 3 stable (Steam Deck) with the videogame Video Horror Society running on Proton 9.

Thanks for this terribly incredible tool btw.
